frontend/src/components/course-search/CourseSearch.tsx

```diff
@@ -13,13 +13,38 @@ import CourseRow from "@components/course-search/CourseRow";
 
 import Css from "./CourseSearch.module.css";
 import { memo, useCallback } from "react";
-import { useActiveSectionsQuery } from "@hooks/section";
+import { useActiveSectionsLookup, useActiveSectionsQuery } from "@hooks/section";
+import { useActiveSchedule } from "@hooks/schedule";
+import { sectionsConflict } from "@lib/schedule";
 
 export default memo(function CourseSearch() {
     const sections: APIv4.Section[] | undefined = useActiveSectionsQuery();
     const searchText = useStore((store) => store.searchText);
     const searchFilters = useStore((store) => store.searchFilters);
     const areas = useCourseAreaDescription().data;
+    const showOnlyNonConflicting = useStore((store) => store.showOnlyNonConflicting);
+    const activeSchedule = useActiveSchedule();
+    const sectionsLookup = useActiveSectionsLookup();
+    const selectedSections: APIv4.Section[] = activeSchedule?.sections
+        .filter((us) => us.attrs.selected)
+        .map((us) =>
+            sectionsLookup.get(APIv4.stringifySectionCodeLong(us.section)),
+        )
+        .filter((s) => s !== undefined) as APIv4.Section[];
+
+    function getNonConflictingSections(selectedSections: APIv4.Section[] | undefined,
+        sections: APIv4.Section[] | undefined
```

Comment:
You don't need the `| undefined` here because neither arguments can be `undefined`, based on what you passed into this function in line 78 (they can, however, be empty arrays).